Add MFA explainer

Fleid opened the pull request:
Following this conversation: https://dbt-labs.slack.com/archives/C02SRNY2EQ4/p1719421396340509

Fleid committed:
Add MFA explainer

Commit:
add img, rephrase, and turn to toggles

Commit:
line break

Commit:
img resize

Commit:
switch format

mirnawong1 commented:
hey @Fleid , this looks awesome, thank you so much for doing this! I'm wondering if this content is better suited under the 'Authentication' header so it correctly represents the other way to authenticate? what do you think?  e.g.:

```
## Authentication methods

...other auth methods...

### Snowflake MFA

**Available in:** Development environments,  Deployment environments

dbt Cloud supports Snowflake's [multi-factor authentication (MFA)](https://docs.snowflake.com/en/user-guide/security-mfa) for increased login security. Snowflake's MFA support is powered by the Duo Security service.

- In dbt Cloud, set the following [extended attribute](/docs/dbt-cloud-environments#extended-attributes) in the development environment **General settings** page, under the **Extended attributes** section:

  ```yaml
  authenticator: username_password_mfa
  ```
- To reduce the number of user prompts when connecting to Snowflake with MFA, [enable token caching](https://docs.snowflake.com/en/user-guide/security-mfa#using-mfa-token-caching-to-minimize-the-number-of-prompts-during-authentication-optional) in Snowflake.
- Optionally, if users miss prompts and their Snowflake accounts get locked, you can prevent automatic retries by adding the following in the same **Extended attributes** section:
  ```yaml
  connect_retries: 0
  ```

<Lightbox src="/img/docs/dbt-cloud/cloud-configuring-dbt-cloud/extended-attributes-mfa.jpg" width="70%" title="Configure the MFA username and password, and connect_retries in the development environment settings." />

mirnawong1 committed:
Merge branch 'current' into snowflake-mfa

Commit:
original

Commit:
Merge branch 'snowflake-mfa' of https://github.com/dbt-labs/docs.getdbt.com into snowflake-mfa

Commit:
reove extra dash

Fleid commented:
@mirnawong1 good catch!

I would not add an entire section, to me it's mostly redundant. MFA is an option of username/password. So I would move the paragraph from the bottom section to under username password. I can do it if you want ;)

mirnawong1 committed:
Merge branch 'current' into snowflake-mfa

mirnawong1 commented:
oh ok i see! happy to do that for you @Fleid ! no worries 🙏

mirnawong1 committed:
Update connect-snowflake.md

mirnawong1 committed:
Merge branch 'current' into snowflake-mfa

mirnawong1 committed:
Merge branch 'current' into snowflake-mfa

matthewshaver committed:
Update connect-snowflake.md

Expanding the original phrasing to include more details about requirements. One request we get frequently is to include what type of personas are required to complete tasks in each environment so the proper teams can be coordinated

matthewshaver commented on website/docs/docs/cloud/connect-data-platform/connect-snowflake.md:
```diff
@@ -33,6 +33,31 @@ to authenticate dbt Cloud to run queries against Snowflake on behalf of a Snowfl
+**Prerequisites:**
+- A Deployment or Development environment in a dbt Cloud project
+- The Duo authentication app
+- Admin access to Snowflake to configure MFA settings for the account
```
```suggestion
- Admin access to Snowflake if MFA  settings have not already been applied to the account
```

Fleid replied:
Can you remove the deployment environment option? I don't want to say it's not supported (because it is), but MFA should really only be used in development

Apart from that, we're golden!

mirnawong1 replied:
will get the removed and then merge!

matthewshaver commented on website/docs/docs/cloud/connect-data-platform/connect-snowflake.md:
```diff
@@ -33,6 +33,31 @@ to authenticate dbt Cloud to run queries against Snowflake on behalf of a Snowfl
+- A Deployment or Development environment in a dbt Cloud project
+- The Duo authentication app
+- Admin access to Snowflake to configure MFA settings for the account
+- Admin access to the environment in dbt Cloud for configuration
```
```suggestion
- [Admin (write) access](/docs/cloud/manage-access/seats-and-users) to the environments in dbt Cloud
```

matthewshaver committed:
Apply suggestions from code review

Fleid committed:
Merge branch 'current' into snowflake-mfa

mirnawong1 committed:
Merge branch 'current' into snowflake-mfa

mirnawong1 commented on website/docs/docs/cloud/connect-data-platform/connect-snowflake.md:
```diff
@@ -33,6 +33,31 @@ to authenticate dbt Cloud to run queries against Snowflake on behalf of a Snowfl
+### Snowflake MFA
+
+**Prerequisites:**
+- A Deployment or Development environment in a dbt Cloud project
```
```suggestion
- A development environment in a dbt Cloud project
```

mirnawong1 committed:
Update website/docs/docs/cloud/connect-data-platform/connect-snowflake.md

mirnawong1 committed:
Update connect-snowflake.md

mirnawong1 committed:
Merge branch 'current' into snowflake-mfa

mirnawong1 commented:
thanks team, let's ship this awesome doc!
